exp/query/api.go

```diff
@@ -121,10 +123,11 @@ func (a *API) Search(ctx context.Context, req *SearchRequest) (*SearchResponse,
 		return nil, fmt.Errorf("tenant %q is not offloaded, %w", req.Tenant, ErrInvalidTenant)
 	}
 
-	store, localPath, err := a.EnsureLSM(ctx, req.Collection, req.Tenant, false)
+	store, localPath, err := a.FetchLSM(ctx, req.Collection, req.Tenant, tenantVersion)
```

Comment:
NIT: Is the previous `EnsureLSM` still used elsewhere ? If not it can be removed.

Reply:
We don't use it anywhere now. Still keeping it because Nate may need it for data versioning. Happy to remove in the follow up PR if we decide not to use it.